File: onyx-database-tests/src/main/java/pojo/ComplexObject.java
package pojo;

import java.io.Serializable;
import java.util.Date;

/**
 * Created by timothy.osborn on 4/14/15.
 */
public class ComplexObject implements Serializable
{
    public ComplexObject()
    {

    }
    public ComplexObject mine;
    public ComplexObjectChild child;

    public Date dateValue = null;

}

File: onyx-database-tests/src/main/java/pojo/ComplexObject.kt
package pojo

import java.io.Serializable
import java.util.Date

/**
 * Created by timothy.osborn on 4/14/15.
 */
class ComplexObject : Serializable {
    var mine: ComplexObject? = null
    var child: ComplexObjectChild? = null

    var dateValue: Date? = null

}
